var/spack/repos/builtin/packages/cdo/package.py

```diff
@@ -25,93 +25,42 @@
-
-    def install(self, spec, prefix):
-        config_args = ["--prefix=" + prefix,
-                       "--enable-shared",
-                       "--enable-static"]
-
-        if '+szip' in spec:
-            config_args.append('--with-szlib=' + spec['szip'].prefix)
-        else:
-            config_args.append('--without-szlib')
-
-        if '+hdf5' in spec:
-            config_args.append('--with-hdf5=' + spec['hdf5'].prefix)
-        else:
-            config_args.append('--without-hdf5')
-
-        if '+netcdf' in spec:
-            config_args.append('--with-netcdf=' + spec['netcdf'].prefix)
-        else:
-            config_args.append('--without-netcdf')
-
-        if '+udunits2' in spec:
-            config_args.append('--with-udunits2=' + spec['udunits2'].prefix)
-        else:
-            config_args.append('--without-udunits2')
-
-        if '+grib' in spec:
-            config_args.append('--with-grib_api=' + spec['grib-api'].prefix)
-        else:
-            config_args.append('--without-grib_api')
-
-        if '+libxml2' in spec:
-            config_args.append('--with-libxml2=' + spec['libxml2'].prefix)
-        else:
-            config_args.append('--without-libxml2')
-
-        if '+proj' in spec:
-            config_args.append('--with-proj=' + spec['proj'].prefix)
-        else:
-            config_args.append('--without-proj')
-
-        if '+curl' in spec:
-            config_args.append('--with-curl=' + spec['curl'].prefix)
-        else:
-            config_args.append('--without-curl')
-
-        if '+fftw' in spec:
-            config_args.append('--with-fftw3')
-        else:
-            config_args.append('--without-fftw3')
-
-        if '+magics' in spec:
-            config_args.append('--with-magics=' + spec['magics'].prefix)
-        else:
-            config_args.append('--without-magics')
-
-        configure(*config_args)
-
-        make()
-        make('install')
+    variant(
+        'enable',
+        default='',
+        description='Enable support for various tools',
+        values=('szip', 'hdf5', 'netcdf', 'udunits2', 'grib', 'libxml2', 'proj', 'curl', 'fftw', 'magics'),  # NOQA: ignore=E501
+        exclusive=False
+    )
+
+    depends_on('szip', when='enable=szip')
+    depends_on('netcdf', when='enable=netcdf')
+    depends_on('hdf5+threadsafe', when='enable=hdf5')
+    depends_on('udunits2', when='enable=udunits2')
+    depends_on('grib-api', when='enable=grib')
+    depends_on('libxml2', when='enable=libxml2')
+    depends_on('proj', when='enable=proj')
+    depends_on('curl', when='enable=curl')
+    depends_on('fftw', when='enable=fftw')
+    depends_on('magics', when='enable=magics')
+
+    def configure_args(self):
+        spec = self.spec
+        config_args = [
+            '--enable-shared',
+            '--enable-static'
+        ]
+        config_args.extend(
+            self.with_or_without('enable', lambda x: spec[x].prefix)
+        )
+        return config_args
```

Comment:
This is cryptic.  We should not expect package authors to use `lambda`.  Understanding this code also requires one know what `with_or_without()` does --- a specialized function only useful for Autotools.  In the end, I think the gain has been minor, if not net negative.